### DISCUSSION TITLE
process multiple messages at the same

### DISCUSSION THREAD

**Discussion by immeyti** [php-amqplib/php-amqplib#899] [Q&A] [answered]
Is it possible that process multiple messages at the same time with just one consumer?

**Comment by ramunasd** [comment]
PHP does not support threads out of the box, if You mean really "at the same time". Even if it does, this library could not handle simultanous packets from different threads.

**Reply by immeyti** [reply]
I want to consume all messages with just one consumer (reuse consumer). Is it possible?
At the moment I have a consumer per message

**Comment by liborm85** [comment, marked as answer]
Yes, consume all messages is possible, this is standard feature.
See demo [/demo/amqp_consumer.php](https://github.com/php-amqplib/php-amqplib/blob/2.12.2/demo/amqp_consumer.php)

This line is consume all messages: [amqp_consumer.php#L68](https://github.com/php-amqplib/php-amqplib/blob/2.12.2/demo/amqp_consumer.php#L68)
And here [amqp_consumer.php#L44-L56](https://github.com/php-amqplib/php-amqplib/blob/2.12.2/demo/amqp_consumer.php#L44-L56) is method for your processing each message.
